base/src/com/google/idea/blaze/base/sync/workspace/ExecutionRootPathResolver.java

```diff
@@ -118,4 +120,24 @@ private static String getFirstPathComponent(String path) {
   private static boolean isExternalWorkspacePath(String firstPathComponent) {
     return firstPathComponent.equals("external");
   }
+
+  /**
+  * Converts external paths in unstable locations under execroot that may change on the next build
+  * to stable ones under their external workspace.
+  * That is, converts paths under <outputBase>/execroot/<workspace_name>/external/ to paths under
+  * <outputBase>/external/.
+  * Safe to call on all paths; non-external paths are left as they are.
+  * See https://github.com/bazelbuild/intellij/issues/2766 for more details.
+  */
+  private static File convertExternalToStable(File f) {
+    String regexSep = Pattern.quote(f.separator);
+    // Workspace name defaults to __main__ per DEFAULT_REPOSITORY_DIRECTORY in 
+    // https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/cmdline/LabelConstants.java
+    // Valid workspace name regex copied from LEGAL_WORKSPACE_NAME in 
```

Comment:
If you all have a good way of referencing Bazel source from within the IntelliJ plugin, please say something. Ideally, it'd be good to reference LEGAL_WORKSPACE_NAME instead of duplicating the regex.

Reply:
I do not think there is a way to use `LEGAL_WORKSPACE_NAME` here.

Reply:
👍🏻 Sounds like we proceed then. Figured I should ask, though :)